tests/sentry/sentry_metrics/test_all_indexers.py

```diff
@@ -342,6 +346,39 @@ def test_already_created_plus_written_results(indexer, indexer_cache, use_case_i
         )
 
 
+def test_invalid_timestamp_in_indexer_cache(indexer, indexer_cache) -> None:
+    """
+    Test that the caching indexer will incur a miss if the data is staying in the cache for too long
+    """
+    with override_options(
+        {
+            "sentry-metrics.indexer.read-new-cache-namespace": True,
+            "sentry-metrics.indexer.write-new-cache-namespace": True,
+        }
+    ):
+        indexer = CachingIndexer(indexer_cache, indexer)
+
+        use_case_id = UseCaseID.SPANS
+        org_id = 1
+        s = "str"
+
+        res = indexer.bulk_record({use_case_id: {org_id: {s}}})
+        id = res.get_mapped_results()[use_case_id][org_id][s]
+        assert res.get_fetch_metadata()[use_case_id][org_id][s].fetch_type == FetchType.FIRST_SEEN
+
+        indexer.cache.cache.set(
+            indexer.cache._make_namespaced_cache_key(
+                BULK_RECORD_CACHE_NAMESPACE, f"{use_case_id.value}:{org_id}:{s}"
+            ),
+            indexer.cache._make_cache_val(id, 0),
+            version=indexer_cache.version,
+        )
+
+        res = indexer.bulk_record({use_case_id: {org_id: {s}}})
+        id = res.get_mapped_results()[use_case_id][org_id][s]
+        assert res.get_fetch_metadata()[use_case_id][org_id][s].fetch_type == FetchType.DB_READ
```

Comment:
This is good, thanks. Is it possible to check that the value within the cache key:value pair contains a new timestamp that is set to a time that is closer to `now()`?